pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -140,6 +176,37 @@ func makeVirtualServiceSpec(ia v1alpha1.IngressAccessor, gateways map[v1alpha1.I
 	return &spec
 }
 
+func makeProbeRoute(host string) v1alpha3.HTTPRoute {
```

Comment:
Why do we need to add a dummy route? Couldn't we also use our probe? It'll either return activator or queue-proxy but we'd know it works without side-effects I think? Is this to keep the layering strict?

Reply:
> Why do we need to add a dummy route?
It is an invalid Spec without it.

> Couldn't we also use our probe? It'll either return activator or queue-proxy but we'd know it works without side-effects I think? Is this to keep the layering strict?
We could. It is simply because it is simpler, has lower latency and more reliable.

Reply:
Is it guaranteed that the fake route is propageted on the same paths as the actual route? As in: Can we guarantee the actual route is ready and propagated to envoys if we probe the fake route?

Reply:
We can double check the Istio implementation.
Even with an actual route, your question still stands:

> Can we guarantee the actual route is ready and propagated to envoys if we probe the **probe** route?

We are assuming that yes, since both routes are bound to the same Istio Gateway, they will be both applied to the same Envoy pods.

An interesting related question is how does Istio apply the config changes?
If it is atomic, good. If not, it is possible that the probe route is applied before the actual route.